Consolidating executor runners in CMake - xnnpack

BujSet opened the pull request:
### Summary
Continuing the process of consolidating the executor runners to top level CMakeLists.txt file. Completing this process for the xnnpack backend in the cmake build flow. 

`Partially Fixes #<10819>` : https://github.com/pytorch/executorch/issues/10819.

### Test plan
Successfully ran the commands in the [building from source](https://github.com/BujSet/executorch/blob/main/docs/source/using-executorch-building-from-source.md) guide. Updated the github actions to not use the generic `executor_runner` and remove the invocation for the backend-specific `xnn_executor_runner`. Also ran the following formatinng and lint checks:

```cmake-format -i backends/xnnpack/CMakeLists.txt```

```cmake-lint backends/xnnpack/CMakeLists.txt```

Includes updates to documentation and tutorials (flows from examples and tutorials were verify to ensure build functionality).

Note: the buck flow side of things was unmodified, and support for that flow remains for a future change.

BujSet committed:
Consolidating executor runners in CMake - xnnpack

Continuing the process of consolidating the executor runners to top
level CMakeLists.txt file. Completing this process for the xnnpack
backend. Change also updates the CI tests to not use the generic
executor_runner and removes the specific xnn_executor_runner.

larryliu0820 commented on backends/xnnpack/CMakeLists.txt:
```diff
@@ -153,7 +132,6 @@ endif()
 
 install(
   TARGETS xnnpack_backend
-  DESTINATION lib
```
Why change this?

BujSet replied:
I ran `cmake-lint` on this file, and it reported `[E1122] Duplicate keyword argument DESTINATION` because this variable is overridden two lines below to what I believe is the correct destination. It's the same error and fix as in the [formatting PR-L496](https://github.com/pytorch/executorch/pull/11230/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aL469).

larryliu0820 commented on .ci/scripts/test_model.sh:
```diff
@@ -150,10 +137,10 @@ test_model_with_xnnpack() {
-    if [[ ! -f ${CMAKE_OUTPUT_DIR}/backends/xnnpack/xnn_executor_runner ]]; then
-      build_cmake_xnn_executor_runner
+    if [[ ! -f ${CMAKE_OUTPUT_DIR}/executor_runner ]]; then
+      build_cmake_executor_runner
```
You will have to make sure `EXECUTORCH_BUILD_XNNPACK` is ON, right? Maybe let `build_cmake_executor_runner` take a backend string and turn the option on accordingly? This can be extended to other backends.

larryliu0820 commented on .ci/scripts/test_model.sh:
```diff
@@ -150,10 +137,10 @@ test_model_with_xnnpack() {
   if [[ "${BUILD_TOOL}" == "buck2" ]]; then
     buck2 run //examples/xnnpack:xnn_executor_runner -- --model_path "${OUTPUT_MODEL_PATH}"
```
Also after this PR the internal (BUCK) behavior is different than CMake. I think it's fine for now but maybe add a TODO here?

BujSet replied:
Added in a comment for it.

BujSet committed:
Added cmake-format and lint check changes

BujSet committed:
Rebuilding on every test invocation to prevent using wrong cached build

BujSet committed:
Removing refs of xnn_executor_runner

BujSet committed:
Updating docs and other refs to xnn_executor_runner

Note, this change only affects the cmake build process. Similar changes
will be necessary for the buck build flow.
